metron-platform/metron-parsers/src/test/java/org/apache/metron/parsers/regex/RegularExpressionsParserTest.java

```diff
@@ -0,0 +1,152 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.metron.parsers.regex;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertTrue;
+
+public class RegularExpressionsParserTest {
+
+  private RegularExpressionsParser regularExpressionsParser;
+  private JSONObject parserConfig;
+
+  @Before
+  public void setUp() throws Exception {
+    regularExpressionsParser = new RegularExpressionsParser();
+  }
+
+  @Test
+  public void testSSHDParse() throws Exception {
+    String message =
+        "<38>Jun 20 15:01:17 deviceName sshd[11672]: Accepted publickey for prod from 22.22.22.22 port 55555 ssh2";
+
+    parserConfig = getJsonConfig(
+        Paths.get("src/test/resources/config/RegularExpressionsParserConfig.json").toString());
```

Comment:
Actually, parser did parse the message. If you look at the raw_message, it is actually the parsed_message. Nowe certainly there is something weird here. Not sure why REPL thinks that parser failed and not sure why REPL is putting the successfully parsed message into raw_message field. As the parser itself has no relation to raw_message field, I think something is wrong with REPL. This is the parsed message extracted from the REPL output. So certainly REPL got this output from parser. The only way it could have got this output from parser is when parser successfully returned from the **parse** methiod.

```
{
    "dst_process_id": "11672",
    "dst_process_name": "sshd",
    "source.type": "regex",
    "device_name": "deviceName",
    "original_string": "<38>Jun 20 15:01:17 deviceName sshd[11672]: Accepted publickey for prod from 22.22.22.22 port 55555 ssh2",
    "event_info": "Accepted publickey",
    "ip_src_port": "55555",
    "dst_user_id": "prod",
    "app_protocol": "ssh2",
    "guid": "edaee82d-02fb-4ec9-9412-5912fa8d4a6f",
    "syslogpriority": "38",
    "timestamp_device_original": "Jun 20 15:01:17",
    "ip_src_addr": "22.22.22.22"
}
```

Regarding changing the configuration to use @Multiline, I will do that.

Reply:
It could be failing because this parser does not add "timestamp" in the parsed json. In our usecase we add timestamp using stellar. I will update the parser to add a default current system timestamp.

Reply:
Thanks @jagdeepsingh2 .  I will try and debug a little further myself too.  I want to make sure there are no incompatibilities between your parser and the newer changes introduced by the `ParserRunner`.  Glad there isn't something obviously stupid that I am doing. :)

Reply:
If the parser does fail to parse a given message, we need to make sure that the error message kicked out to the error topic has a helpful message, stack trace, etc.  Otherwise, it will be impossible for a user to determine why the parser failed to parse the message. 

While adding the timestamp is probably a good addition,  I don't know that it really solves the problem here.  Right now, I don't really know if the problem is in your parser or in the parser infrastructure, but it is something that I want to make sure we track down.

Reply:
>  I will update the parser to add a default current system timestamp.

Should the timestamp come from system time or should it come from the syslog timestamp?  The latter seems more correct to me.

Reply:
Hi @jagdeepsingh2 - I was able to get this up and running in a debugger.  Your parser will not parse messages successfully after the changes made in #1213. You are likely using this on an older version of Metron.

The parser must produce a JSONObject that contains both a `timestamp` and `original_string` field based on the [validation performed here.](https://github.com/apache/metron/blob/2ee6cc7e0b448d8d27f56f873e2c15a603c53917/metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/BasicParser.java#L34-L46)
 
If you add the timestamp like you mentioned it should work.

Reply:
I [opened this JIRA ](https://issues.apache.org/jira/browse/METRON-1926)to fix the parsing infrastructure.  The error message produced should have made it clear that the message failed because it was missing a timestamp, but it does not.

Reply:
When writing a new parser, it is important that you also implement the integration tests.  An example of a parser submittal that does this is : https://github.com/apache/metron/pull/1279

Reply:
Yes, good point @ottobackwards .

@jagdeepsingh2 - He is referring specifically to the class `Syslog3164ParserIntegrationTest` in that PR.  Should be fairly simple to put together with what you already have.

Reply:
The integration tests have a standard setup.  You have to do a few things, off the top of my head they are ( again check the pr and that parser for details):

- write the IntegrationTest that derives from the base 
- create a default sample configuration for your parser and put it in the configuration area
- add in the raw and parsed data in the integration testing module data directory for comparison

Reply:
Ok, this pr is actually simpler:  https://github.com/apache/metron/pull/1175

Reply:
I have added the timestamp field to the parser and also have added the more targeted configuration using @Multiline now.

Will try to add Integration tests as well.